dagshub/upload/wrapper.py

```diff
@@ -281,6 +290,11 @@ def upload_files(
                     "new_branch_name": new_branch,
                 }
             )
+            # Drop the upload revision because we're sure this is a new branch
+            self._last_upload_revision = None
```

Comment:
Actually this isn't how I would do it.
Let's look at this scenario:

```python
repo = Repo('a', 'b', branch='c')
repo.upload_files(...)
repo.upload_files(..., new_branch='d')
repo.upload_files(...)
```

The 3rd upload_files can still benefit from the `_last_upload_revision`

Instead, and I think that would also be more readable - if the `new_branch` is passed, then totally skip all the checks and calls to `self.poll_mirror` etc.

Then the repo object is still good for more uploads, and it should be safe as you're working on a totally new branch